docs(auth): add note about using user token vs uid (#5428)
invertase/react-native-firebase: docs/auth/usage/index.md

In case any errors occur, the module provides support for identifying what specifically went wrong by attaching a code
to the error. For a full list of error codes available, view the [Firebase documentation](https://firebase.google.com/docs/reference/js/firebase.auth.Auth#error-codes_3).

## Authenticate with backend server

> To authenticate with the backend server, the user token should be used. This value can be fetched by using the getIdToken method as shown in the following snippet. As mentioned in the [Firebase documentation](https://firebase.google.com/docs/auth/web/manage-users), the user's UID per se should not be used for purposes of authentication. 
> ```
>      import auth from '@react-native-firebase/auth';
>      const user = auth().currentUser;
>      user
>         .getIdToken()
>         .then(token =>
>           console.log('uid token is ', token),
>         ); 
> ```
> 

Reviewer: ```suggestion
The user's token should be used for authentication with your backend systems. The token is fetched with the [getIdToken](https://rnfirebase.io/reference/auth/user#getIdToken) method as shown in the following snippet. As mentioned in the [Firebase documentation](https://firebase.google.com/docs/auth/web/manage-users#get_a_users_profile), the uid should not be used for authentication. 
```
     import auth from '@react-native-firebase/auth';
     const user = auth().currentUser;
     user
        .getIdToken()
        .then(token =>
          console.log('user token is ', token),
        ); 
```
```

In case any errors occur, the module provides support for identifying what specifically went wrong by attaching a code
to the error. For a full list of error codes available, view the [Firebase documentation](https://firebase.google.com/docs/reference/js/firebase.auth.Auth#error-codes_3).

> ## Authenticate with backend server
> The user's token should be used for authentication with your backend systems. The token is fetched with the [getIdToken](https://rnfirebase.io/reference/auth/user#getIdToken) method. As mentioned in the [Firebase documentation](https://firebase.google.com/docs/auth/web/manage-users#get_a_users_profile), the uid should not be used for authentication. 

Reviewer: I believe prettier requires a blank line here to pass lint check, going to try it

```suggestion
## Authenticate with backend server

The user's token should be used for authentication with your backend systems. The token is fetched with the [getIdToken](https://rnfirebase.io/reference/auth/user#getIdToken) method. As mentioned in the [Firebase documentation](https://firebase.google.com/docs/auth/web/manage-users#get_a_users_profile), the uid should not be used for authentication. 
```
